pkg/reconciler/v1alpha1/taskrun/taskrun_test.go

```diff
@@ -1476,6 +1476,41 @@ func TestReconcileTimeouts(t *testing.T) {
 	}
 }
 
+func TestSortTaskRunSteps(t *testing.T) {
+	task := tb.Task("failing-task", "default", tb.TaskSpec(
+		tb.Step("hello", "busybox",
+			tb.Command("/bin/sh"), tb.Args("-c", "echo hello"),
+		),
+		tb.Step("exit", "busybox",
+			tb.Command("/bin/sh"), tb.Args("-c", "exit 1"),
+		),
+		tb.Step("world", "busybox",
+			tb.Command("/bin/sh"), tb.Args("-c", "sleep 30s"),
+		),
+	))
+
+	taskRunStatusSteps := []v1alpha1.StepState {{
+			Name: "world",
+		}, {
+			Name: "exit",
+		}, {
+			Name: "nop",
+		}, {
+			Name: "hello",
+		},
+	}
+
+	sortTaskRunSteps(taskRunStatusSteps, task.Spec.Steps)
+
+	for index, step := range task.Spec.Steps {
```

Comment:
I wonder if it would be more clear to do the `cmp.Diff` on the `taskRunStatusSteps` and an `expectedTaskRunStatusSteps` instead :angel:

Reply:
My question is how to create the fake taskRunStatusSteps and the expectedTaskRunStatusSteps?

Reply:
well, `expectedTaskRunStatusStep` would be declared as `taskRunStatusSteps` but in the right order :angel:

Reply:
Got it.